core/import.go

```diff
@@ -133,15 +133,18 @@ func (hf *Hoverfly) ImportFromURL(url string) error {
 }
 
 // importRequestResponsePairViews - a function to save given pairs into the database.
-func (hf *Hoverfly) importRequestResponsePairViews(pairViews []v2.RequestMatcherResponsePairViewV6) v2.SimulationImportResult {
+func (hf *Hoverfly) importRequestResponsePairViews(pairViews []v2.RequestMatcherResponsePairViewV6) (v2.SimulationImportResult, error) {
 	importResult := v2.SimulationImportResult{}
 	initialStates := map[string]string{}
 	if len(pairViews) > 0 {
 		success := 0
 		failed := 0
 		for i, pairView := range pairViews {
 
-			pair := models.NewRequestMatcherResponsePairFromView(&pairView)
+			pair, err := models.NewRequestMatcherResponsePairFromView(&pairView)
```

Comment:
Why not set the error here , then we don't need to change the signature of `importRequestResponsePairViews` : 
```
pair, err := models.NewRequestMatcherResponsePairFromView(&pairView)
			if err != nil {
				importResult.SetError(err)
				return importResult
			}
```